administrator/components/com_content/models/article.php

```diff
@@ -706,6 +705,70 @@ public function featured($pks, $value = 0)
 	}
 
 	/**
+	 * Method to store the token generated.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since 3.7
+	 */
+	private function shareTokenGenerate()
+	{
+		jimport('joomla.user.helper');
+		$token = JUserHelper::genRandomPassword(16);
+
+		return $token;
+	}
+
+	/**
+	 * Method to store the token generated.
+	 *
+	 * @param   string  $title  The title of the shared article.
+	 *
+	 * @return  token.
+	 *
+	 * @since   3.7
+	 */
+	public function shareToken($title)
+	{
+		$table = $this->getTable('Share', 'ContentTable');
+        	$db = $this->getDbo();
+        	$query = $db->getQuery(true);
+
+        	$query
+             		->select('*')
+             		->from($db->quoteName('#__share_draft'))
+             		->where($db->quoteName('articleId') . '=' . $db->quote($this->get('id')));
+
+       		$db->execute();
+       		$num_rows = $db->getNumRows();
+
+       		if ($num_rows !== null)
+       		{
+			$db = $this->getDbo();
+            		$query = $db->getQuery(true);
+
+            		$query
+             			->select('sharetoken')
+             			->from($db->quoteName('#__share_draft'))
+             			->where($db->quoteName('id') . ' = ' . $db->quote($this->get('id')));
+
+           		$db->execute();
+           		$valueshow = $db->loadObject();
+           		echo "not a new value, the existing value is:" . $valueshow;
```

Comment:
you need to return the existing value here. return $valueshow; (which is a pretty bad variable name)